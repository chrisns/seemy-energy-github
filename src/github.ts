import { Octokit } from '@octokit/rest'
import { createAppAuth } from '@octokit/auth-app'
import { formatPullRequest, formatIssue } from './formatter'
const { paginateRest } = require('@octokit/plugin-paginate-rest')
import { Endpoints } from '@octokit/types'

import SQS from 'aws-sdk/clients/sqs'

import DynamoDB from 'aws-sdk/clients/dynamodb'

export function getAuthenticatedOctokit (installationId: number): Octokit {
  return new Octokit({
    authStrategy: createAppAuth,
    auth: {
      appId: process.env.appId,
      type: 'app',
      privateKey: process.env.privateKey,
      installationId: installationId,
    },
  }).plugin(paginateRest)
}

interface sqsPullMessage {
  owner: string
  repo: string
  pull_number: number
  installation_id: number
}

export async function queryRepo (owner: string, repo: string, octokit: Octokit) {
  for await (const pulls of octokit.paginate.iterator(octokit.pulls.list, {
    owner: owner,
    repo: repo,
  })) {
    return await pulls.data.map(pullListPRtoSQS)
  }
}

export async function pullListPRtoSQS (
  pull: Endpoints['GET /repos/{owner}/{repo}/pulls']['response']['data'][0],
): Promise<SQS.SendMessageResult> {
  return sqsClient
    .sendMessage({
      MessageBody: JSON.stringify(<sqsPullMessage>{
        owner: pull.base.user.login,
        pull_number: pull.number,
        repo: pull.base.repo.url,
        installation_id: 1234,
      }),
      MessageDeduplicationId: `${pull.base.user.login}/${pull.base.repo.url}/${pull.number}`,
      MessageGroupId: `${pull.base.user.login}/${pull.base.repo.url}`,
      QueueUrl: process.env.PR_QUEUE,
    })
    .promise()
}

const config = {
  convertEmptyValues: true,
  ...(process.env.MOCK_DYNAMODB_ENDPOINT && {
    endpoint: process.env.MOCK_DYNAMODB_ENDPOINT,
    sslEnabled: false,
    region: 'local',
  }),
}

export const documentClient = new DynamoDB.DocumentClient(config)

export const sqsClient = new SQS()

const upsert_table = async (payload, table) =>
  documentClient
    .put({
      Item: payload,
      TableName: table,
    })
    .promise()

export async function ETLPullRequest (
  owner: string,
  repo: string,
  pull_number: number,
  octokit: Octokit,
): Promise<void> {
  const pull = await octokit.pull.get({ owner, repo, pull_number })
  const formatted = formatPullRequest(pull)
  await upsert_table(formatted, 'pull')
  return
}

export async function ETLIssue (owner: string, repo: string, issue_number: number, octokit: Octokit): Promise<void> {
  const issue = await octokit.issue.get({ owner, repo, issue_number })
  const formatted = formatIssue(issue)
  await upsert_table(formatted, 'issue')
  return
}
