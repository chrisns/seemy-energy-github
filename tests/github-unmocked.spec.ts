import * as mod from '../src/github'

test('getAuthenticatedOctokit', () => expect(mod.getAuthenticatedOctokit(1234)).toMatchSnapshot())
