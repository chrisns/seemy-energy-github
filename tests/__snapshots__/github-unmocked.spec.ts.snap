// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getAuthenticatedOctokit 1`] = `
_a {
  "actions": Object {
    "addSelectedRepoToOrgSecret": [Function],
    "cancelWorkflowRun": [Function],
    "createOrUpdateOrgSecret": [Function],
    "createOrUpdateRepoSecret": [Function],
    "createRegistrationTokenForOrg": [Function],
    "createRegistrationTokenForRepo": [Function],
    "createRemoveTokenForOrg": [Function],
    "createRemoveTokenForRepo": [Function],
    "createWorkflowDispatch": [Function],
    "deleteArtifact": [Function],
    "deleteOrgSecret": [Function],
    "deleteRepoSecret": [Function],
    "deleteSelfHostedRunnerFromOrg": [Function],
    "deleteSelfHostedRunnerFromRepo": [Function],
    "deleteWorkflowRun": [Function],
    "deleteWorkflowRunLogs": [Function],
    "disableSelectedRepositoryGithubActionsOrganization": [Function],
    "disableWorkflow": [Function],
    "downloadArtifact": [Function],
    "downloadJobLogsForWorkflowRun": [Function],
    "downloadWorkflowRunLogs": [Function],
    "enableSelectedRepositoryGithubActionsOrganization": [Function],
    "enableWorkflow": [Function],
    "getAllowedActionsOrganization": [Function],
    "getAllowedActionsRepository": [Function],
    "getArtifact": [Function],
    "getGithubActionsPermissionsOrganization": [Function],
    "getGithubActionsPermissionsRepository": [Function],
    "getJobForWorkflowRun": [Function],
    "getOrgPublicKey": [Function],
    "getOrgSecret": [Function],
    "getRepoPermissions": [Function],
    "getRepoPublicKey": [Function],
    "getRepoSecret": [Function],
    "getSelfHostedRunnerForOrg": [Function],
    "getSelfHostedRunnerForRepo": [Function],
    "getWorkflow": [Function],
    "getWorkflowRun": [Function],
    "getWorkflowRunUsage": [Function],
    "getWorkflowUsage": [Function],
    "listArtifactsForRepo": [Function],
    "listJobsForWorkflowRun": [Function],
    "listOrgSecrets": [Function],
    "listRepoSecrets": [Function],
    "listRepoWorkflows": [Function],
    "listRunnerApplicationsForOrg": [Function],
    "listRunnerApplicationsForRepo": [Function],
    "listSelectedReposForOrgSecret": [Function],
    "listSelectedRepositoriesEnabledGithubActionsOrganization": [Function],
    "listSelfHostedRunnersForOrg": [Function],
    "listSelfHostedRunnersForRepo": [Function],
    "listWorkflowRunArtifacts": [Function],
    "listWorkflowRuns": [Function],
    "listWorkflowRunsForRepo": [Function],
    "reRunWorkflow": [Function],
    "removeSelectedRepoFromOrgSecret": [Function],
    "setAllowedActionsOrganization": [Function],
    "setAllowedActionsRepository": [Function],
    "setGithubActionsPermissionsOrganization": [Function],
    "setGithubActionsPermissionsRepository": [Function],
    "setSelectedReposForOrgSecret": [Function],
    "setSelectedRepositoriesEnabledGithubActionsOrganization": [Function],
  },
  "activity": Object {
    "checkRepoIsStarredByAuthenticatedUser": [Function],
    "deleteRepoSubscription": [Function],
    "deleteThreadSubscription": [Function],
    "getFeeds": [Function],
    "getRepoSubscription": [Function],
    "getThread": [Function],
    "getThreadSubscriptionForAuthenticatedUser": [Function],
    "listEventsForAuthenticatedUser": [Function],
    "listNotificationsForAuthenticatedUser": [Function],
    "listOrgEventsForAuthenticatedUser": [Function],
    "listPublicEvents": [Function],
    "listPublicEventsForRepoNetwork": [Function],
    "listPublicEventsForUser": [Function],
    "listPublicOrgEvents": [Function],
    "listReceivedEventsForUser": [Function],
    "listReceivedPublicEventsForUser": [Function],
    "listRepoEvents": [Function],
    "listRepoNotificationsForAuthenticatedUser": [Function],
    "listReposStarredByAuthenticatedUser": [Function],
    "listReposStarredByUser": [Function],
    "listReposWatchedByUser": [Function],
    "listStargazersForRepo": [Function],
    "listWatchedReposForAuthenticatedUser": [Function],
    "listWatchersForRepo": [Function],
    "markNotificationsAsRead": [Function],
    "markRepoNotificationsAsRead": [Function],
    "markThreadAsRead": [Function],
    "setRepoSubscription": [Function],
    "setThreadSubscription": [Function],
    "starRepoForAuthenticatedUser": [Function],
    "unstarRepoForAuthenticatedUser": [Function],
  },
  "apps": Object {
    "addRepoToInstallation": [Function],
    "checkToken": [Function],
    "createContentAttachment": [Function],
    "createFromManifest": [Function],
    "createInstallationAccessToken": [Function],
    "deleteAuthorization": [Function],
    "deleteInstallation": [Function],
    "deleteToken": [Function],
    "getAuthenticated": [Function],
    "getBySlug": [Function],
    "getInstallation": [Function],
    "getOrgInstallation": [Function],
    "getRepoInstallation": [Function],
    "getSubscriptionPlanForAccount": [Function],
    "getSubscriptionPlanForAccountStubbed": [Function],
    "getUserInstallation": [Function],
    "getWebhookConfigForApp": [Function],
    "listAccountsForPlan": [Function],
    "listAccountsForPlanStubbed": [Function],
    "listInstallationReposForAuthenticatedUser": [Function],
    "listInstallations": [Function],
    "listInstallationsForAuthenticatedUser": [Function],
    "listPlans": [Function],
    "listPlansStubbed": [Function],
    "listReposAccessibleToInstallation": [Function],
    "listSubscriptionsForAuthenticatedUser": [Function],
    "listSubscriptionsForAuthenticatedUserStubbed": [Function],
    "removeRepoFromInstallation": [Function],
    "resetToken": [Function],
    "revokeInstallationAccessToken": [Function],
    "scopeToken": [Function],
    "suspendInstallation": [Function],
    "unsuspendInstallation": [Function],
    "updateWebhookConfigForApp": [Function],
  },
  "auth": [Function],
  "billing": Object {
    "getGithubActionsBillingOrg": [Function],
    "getGithubActionsBillingUser": [Function],
    "getGithubPackagesBillingOrg": [Function],
    "getGithubPackagesBillingUser": [Function],
    "getSharedStorageBillingOrg": [Function],
    "getSharedStorageBillingUser": [Function],
  },
  "checks": Object {
    "create": [Function],
    "createSuite": [Function],
    "get": [Function],
    "getSuite": [Function],
    "listAnnotations": [Function],
    "listForRef": [Function],
    "listForSuite": [Function],
    "listSuitesForRef": [Function],
    "rerequestSuite": [Function],
    "setSuitesPreferences": [Function],
    "update": [Function],
  },
  "codeScanning": Object {
    "getAlert": [Function],
    "listAlertsForRepo": [Function],
    "listRecentAnalyses": [Function],
    "updateAlert": [Function],
    "uploadSarif": [Function],
  },
  "codesOfConduct": Object {
    "getAllCodesOfConduct": [Function],
    "getConductCode": [Function],
    "getForRepo": [Function],
  },
  "emojis": Object {
    "get": [Function],
  },
  "enterpriseAdmin": Object {
    "disableSelectedOrganizationGithubActionsEnterprise": [Function],
    "enableSelectedOrganizationGithubActionsEnterprise": [Function],
    "getAllowedActionsEnterprise": [Function],
    "getGithubActionsPermissionsEnterprise": [Function],
    "listSelectedOrganizationsEnabledGithubActionsEnterprise": [Function],
    "setAllowedActionsEnterprise": [Function],
    "setGithubActionsPermissionsEnterprise": [Function],
    "setSelectedOrganizationsEnabledGithubActionsEnterprise": [Function],
  },
  "gists": Object {
    "checkIsStarred": [Function],
    "create": [Function],
    "createComment": [Function],
    "delete": [Function],
    "deleteComment": [Function],
    "fork": [Function],
    "get": [Function],
    "getComment": [Function],
    "getRevision": [Function],
    "list": [Function],
    "listComments": [Function],
    "listCommits": [Function],
    "listForUser": [Function],
    "listForks": [Function],
    "listPublic": [Function],
    "listStarred": [Function],
    "star": [Function],
    "unstar": [Function],
    "update": [Function],
    "updateComment": [Function],
  },
  "git": Object {
    "createBlob": [Function],
    "createCommit": [Function],
    "createRef": [Function],
    "createTag": [Function],
    "createTree": [Function],
    "deleteRef": [Function],
    "getBlob": [Function],
    "getCommit": [Function],
    "getRef": [Function],
    "getTag": [Function],
    "getTree": [Function],
    "listMatchingRefs": [Function],
    "updateRef": [Function],
  },
  "gitignore": Object {
    "getAllTemplates": [Function],
    "getTemplate": [Function],
  },
  "graphql": [Function],
  "hook": [Function],
  "interactions": Object {
    "getRestrictionsForAuthenticatedUser": [Function],
    "getRestrictionsForOrg": [Function],
    "getRestrictionsForRepo": [Function],
    "getRestrictionsForYourPublicRepos": [Function],
    "removeRestrictionsForAuthenticatedUser": [Function],
    "removeRestrictionsForOrg": [Function],
    "removeRestrictionsForRepo": [Function],
    "removeRestrictionsForYourPublicRepos": [Function],
    "setRestrictionsForAuthenticatedUser": [Function],
    "setRestrictionsForOrg": [Function],
    "setRestrictionsForRepo": [Function],
    "setRestrictionsForYourPublicRepos": [Function],
  },
  "issues": Object {
    "addAssignees": [Function],
    "addLabels": [Function],
    "checkUserCanBeAssigned": [Function],
    "create": [Function],
    "createComment": [Function],
    "createLabel": [Function],
    "createMilestone": [Function],
    "deleteComment": [Function],
    "deleteLabel": [Function],
    "deleteMilestone": [Function],
    "get": [Function],
    "getComment": [Function],
    "getEvent": [Function],
    "getLabel": [Function],
    "getMilestone": [Function],
    "list": [Function],
    "listAssignees": [Function],
    "listComments": [Function],
    "listCommentsForRepo": [Function],
    "listEvents": [Function],
    "listEventsForRepo": [Function],
    "listEventsForTimeline": [Function],
    "listForAuthenticatedUser": [Function],
    "listForOrg": [Function],
    "listForRepo": [Function],
    "listLabelsForMilestone": [Function],
    "listLabelsForRepo": [Function],
    "listLabelsOnIssue": [Function],
    "listMilestones": [Function],
    "lock": [Function],
    "removeAllLabels": [Function],
    "removeAssignees": [Function],
    "removeLabel": [Function],
    "setLabels": [Function],
    "unlock": [Function],
    "update": [Function],
    "updateComment": [Function],
    "updateLabel": [Function],
    "updateMilestone": [Function],
  },
  "licenses": Object {
    "get": [Function],
    "getAllCommonlyUsed": [Function],
    "getForRepo": [Function],
  },
  "log": Object {
    "Console": [Function],
    "_counters": Object {},
    "_formatBuffer": [Function],
    "_groupDepth": 0,
    "_stderr": Socket {
      "_destroy": [Function],
      "_events": Object {
        "end": [Function],
      },
      "_eventsCount": 1,
      "_hadError": false,
      "_host": null,
      "_isStdio": true,
      "_maxListeners": undefined,
      "_parent": null,
      "_pendingData": null,
      "_pendingEncoding": "",
      "_readableState": ReadableState {
        "autoDestroy": false,
        "awaitDrainWriters": null,
        "buffer": BufferList {
          "head": null,
          "length": 0,
          "tail": null,
        },
        "closeEmitted": false,
        "closed": false,
        "decoder": null,
        "defaultEncoding": "utf8",
        "destroyed": false,
        "emitClose": false,
        "emittedReadable": false,
        "encoding": null,
        "endEmitted": false,
        "ended": false,
        "errorEmitted": false,
        "errored": null,
        "flowing": null,
        "highWaterMark": 16384,
        "length": 0,
        "multiAwaitDrain": false,
        "needReadable": false,
        "objectMode": false,
        "pipes": Array [],
        "readable": false,
        "readableListening": false,
        "reading": false,
        "readingMore": false,
        "resumeScheduled": false,
        "sync": true,
        Symbol(kPaused): null,
      },
      "_server": null,
      "_sockname": null,
      "_type": "pipe",
      "_writableState": WritableState {
        "afterWriteTickInfo": null,
        "allBuffers": true,
        "allNoop": true,
        "autoDestroy": false,
        "bufferProcessing": false,
        "buffered": Array [],
        "bufferedIndex": 0,
        "closeEmitted": false,
        "closed": false,
        "corked": 0,
        "decodeStrings": false,
        "defaultEncoding": "utf8",
        "destroyed": false,
        "emitClose": false,
        "ended": false,
        "ending": false,
        "errorEmitted": false,
        "errored": null,
        "finalCalled": false,
        "finished": false,
        "highWaterMark": 16384,
        "length": 0,
        "needDrain": false,
        "objectMode": false,
        "onwrite": [Function],
        "pendingcb": 0,
        "prefinished": false,
        "sync": true,
        "writecb": null,
        "writelen": 0,
        "writing": false,
      },
      "allowHalfOpen": false,
      "connecting": false,
      "destroySoon": [Function],
      "fd": 1,
      "server": null,
      Symbol(async_id_symbol): 6,
      Symbol(kHandle): Pipe {
        Symbol(owner_symbol): [Circular],
      },
      Symbol(kSetNoDelay): false,
      Symbol(lastWriteQueueSize): 0,
      Symbol(timeout): null,
      Symbol(kBuffer): null,
      Symbol(kBufferCb): null,
      Symbol(kBufferGen): null,
      Symbol(kCapture): false,
      Symbol(kBytesRead): 0,
      Symbol(kBytesWritten): 0,
    },
    "_stdout": Socket {
      "_destroy": [Function],
      "_events": Object {
        "end": [Function],
      },
      "_eventsCount": 1,
      "_hadError": false,
      "_host": null,
      "_isStdio": true,
      "_maxListeners": undefined,
      "_parent": null,
      "_pendingData": null,
      "_pendingEncoding": "",
      "_readableState": ReadableState {
        "autoDestroy": false,
        "awaitDrainWriters": null,
        "buffer": BufferList {
          "head": null,
          "length": 0,
          "tail": null,
        },
        "closeEmitted": false,
        "closed": false,
        "decoder": null,
        "defaultEncoding": "utf8",
        "destroyed": false,
        "emitClose": false,
        "emittedReadable": false,
        "encoding": null,
        "endEmitted": false,
        "ended": false,
        "errorEmitted": false,
        "errored": null,
        "flowing": null,
        "highWaterMark": 16384,
        "length": 0,
        "multiAwaitDrain": false,
        "needReadable": false,
        "objectMode": false,
        "pipes": Array [],
        "readable": false,
        "readableListening": false,
        "reading": false,
        "readingMore": false,
        "resumeScheduled": false,
        "sync": true,
        Symbol(kPaused): null,
      },
      "_server": null,
      "_sockname": null,
      "_type": "pipe",
      "_writableState": WritableState {
        "afterWriteTickInfo": null,
        "allBuffers": true,
        "allNoop": true,
        "autoDestroy": false,
        "bufferProcessing": false,
        "buffered": Array [],
        "bufferedIndex": 0,
        "closeEmitted": false,
        "closed": false,
        "corked": 0,
        "decodeStrings": false,
        "defaultEncoding": "utf8",
        "destroyed": false,
        "emitClose": false,
        "ended": false,
        "ending": false,
        "errorEmitted": false,
        "errored": null,
        "finalCalled": false,
        "finished": false,
        "highWaterMark": 16384,
        "length": 0,
        "needDrain": false,
        "objectMode": false,
        "onwrite": [Function],
        "pendingcb": 0,
        "prefinished": false,
        "sync": true,
        "writecb": null,
        "writelen": 0,
        "writing": false,
      },
      "allowHalfOpen": false,
      "connecting": false,
      "destroySoon": [Function],
      "fd": 1,
      "server": null,
      Symbol(async_id_symbol): 6,
      Symbol(kHandle): Pipe {
        Symbol(owner_symbol): [Circular],
      },
      Symbol(kSetNoDelay): false,
      Symbol(lastWriteQueueSize): 0,
      Symbol(timeout): null,
      Symbol(kBuffer): null,
      Symbol(kBufferCb): null,
      Symbol(kBufferGen): null,
      Symbol(kCapture): false,
      Symbol(kBytesRead): 0,
      Symbol(kBytesWritten): 0,
    },
    "_timers": Object {},
    "assert": [Function],
    "clear": [Function],
    "count": [Function],
    "countReset": [Function],
    "debug": [Function],
    "dir": [Function],
    "dirxml": [Function],
    "error": [Function],
    "group": [Function],
    "groupCollapsed": [Function],
    "groupEnd": [Function],
    "info": [Function],
    "log": [Function],
    "table": [Function],
    "time": [Function],
    "timeEnd": [Function],
    "timeLog": [Function],
    "trace": [Function],
    "warn": [Function],
  },
  "markdown": Object {
    "render": [Function],
    "renderRaw": [Function],
  },
  "meta": Object {
    "get": [Function],
    "getOctocat": [Function],
    "getZen": [Function],
    "root": [Function],
  },
  "migrations": Object {
    "cancelImport": [Function],
    "deleteArchiveForAuthenticatedUser": [Function],
    "deleteArchiveForOrg": [Function],
    "downloadArchiveForOrg": [Function],
    "getArchiveForAuthenticatedUser": [Function],
    "getCommitAuthors": [Function],
    "getImportStatus": [Function],
    "getLargeFiles": [Function],
    "getStatusForAuthenticatedUser": [Function],
    "getStatusForOrg": [Function],
    "listForAuthenticatedUser": [Function],
    "listForOrg": [Function],
    "listReposForOrg": [Function],
    "listReposForUser": [Function],
    "mapCommitAuthor": [Function],
    "setLfsPreference": [Function],
    "startForAuthenticatedUser": [Function],
    "startForOrg": [Function],
    "startImport": [Function],
    "unlockRepoForAuthenticatedUser": [Function],
    "unlockRepoForOrg": [Function],
    "updateImport": [Function],
  },
  "orgs": Object {
    "blockUser": [Function],
    "cancelInvitation": [Function],
    "checkBlockedUser": [Function],
    "checkMembershipForUser": [Function],
    "checkPublicMembershipForUser": [Function],
    "convertMemberToOutsideCollaborator": [Function],
    "createInvitation": [Function],
    "createWebhook": [Function],
    "deleteWebhook": [Function],
    "get": [Function],
    "getMembershipForAuthenticatedUser": [Function],
    "getMembershipForUser": [Function],
    "getWebhook": [Function],
    "getWebhookConfigForOrg": [Function],
    "list": [Function],
    "listAppInstallations": [Function],
    "listBlockedUsers": [Function],
    "listFailedInvitations": [Function],
    "listForAuthenticatedUser": [Function],
    "listForUser": [Function],
    "listInvitationTeams": [Function],
    "listMembers": [Function],
    "listMembershipsForAuthenticatedUser": [Function],
    "listOutsideCollaborators": [Function],
    "listPendingInvitations": [Function],
    "listPublicMembers": [Function],
    "listWebhooks": [Function],
    "pingWebhook": [Function],
    "removeMember": [Function],
    "removeMembershipForUser": [Function],
    "removeOutsideCollaborator": [Function],
    "removePublicMembershipForAuthenticatedUser": [Function],
    "setMembershipForUser": [Function],
    "setPublicMembershipForAuthenticatedUser": [Function],
    "unblockUser": [Function],
    "update": [Function],
    "updateMembershipForAuthenticatedUser": [Function],
    "updateWebhook": [Function],
    "updateWebhookConfigForOrg": [Function],
  },
  "paginate": [Function],
  "projects": Object {
    "addCollaborator": [Function],
    "createCard": [Function],
    "createColumn": [Function],
    "createForAuthenticatedUser": [Function],
    "createForOrg": [Function],
    "createForRepo": [Function],
    "delete": [Function],
    "deleteCard": [Function],
    "deleteColumn": [Function],
    "get": [Function],
    "getCard": [Function],
    "getColumn": [Function],
    "getPermissionForUser": [Function],
    "listCards": [Function],
    "listCollaborators": [Function],
    "listColumns": [Function],
    "listForOrg": [Function],
    "listForRepo": [Function],
    "listForUser": [Function],
    "moveCard": [Function],
    "moveColumn": [Function],
    "removeCollaborator": [Function],
    "update": [Function],
    "updateCard": [Function],
    "updateColumn": [Function],
  },
  "pulls": Object {
    "checkIfMerged": [Function],
    "create": [Function],
    "createReplyForReviewComment": [Function],
    "createReview": [Function],
    "createReviewComment": [Function],
    "deletePendingReview": [Function],
    "deleteReviewComment": [Function],
    "dismissReview": [Function],
    "get": [Function],
    "getReview": [Function],
    "getReviewComment": [Function],
    "list": [Function],
    "listCommentsForReview": [Function],
    "listCommits": [Function],
    "listFiles": [Function],
    "listRequestedReviewers": [Function],
    "listReviewComments": [Function],
    "listReviewCommentsForRepo": [Function],
    "listReviews": [Function],
    "merge": [Function],
    "removeRequestedReviewers": [Function],
    "requestReviewers": [Function],
    "submitReview": [Function],
    "update": [Function],
    "updateBranch": [Function],
    "updateReview": [Function],
    "updateReviewComment": [Function],
  },
  "rateLimit": Object {
    "get": [Function],
  },
  "reactions": Object {
    "createForCommitComment": [Function],
    "createForIssue": [Function],
    "createForIssueComment": [Function],
    "createForPullRequestReviewComment": [Function],
    "createForTeamDiscussionCommentInOrg": [Function],
    "createForTeamDiscussionInOrg": [Function],
    "deleteForCommitComment": [Function],
    "deleteForIssue": [Function],
    "deleteForIssueComment": [Function],
    "deleteForPullRequestComment": [Function],
    "deleteForTeamDiscussion": [Function],
    "deleteForTeamDiscussionComment": [Function],
    "deleteLegacy": [Function],
    "listForCommitComment": [Function],
    "listForIssue": [Function],
    "listForIssueComment": [Function],
    "listForPullRequestReviewComment": [Function],
    "listForTeamDiscussionCommentInOrg": [Function],
    "listForTeamDiscussionInOrg": [Function],
  },
  "repos": Object {
    "acceptInvitation": [Function],
    "addAppAccessRestrictions": [Function],
    "addCollaborator": [Function],
    "addStatusCheckContexts": [Function],
    "addTeamAccessRestrictions": [Function],
    "addUserAccessRestrictions": [Function],
    "checkCollaborator": [Function],
    "checkVulnerabilityAlerts": [Function],
    "compareCommits": [Function],
    "createCommitComment": [Function],
    "createCommitSignatureProtection": [Function],
    "createCommitStatus": [Function],
    "createDeployKey": [Function],
    "createDeployment": [Function],
    "createDeploymentStatus": [Function],
    "createDispatchEvent": [Function],
    "createForAuthenticatedUser": [Function],
    "createFork": [Function],
    "createInOrg": [Function],
    "createOrUpdateFileContents": [Function],
    "createPagesSite": [Function],
    "createRelease": [Function],
    "createUsingTemplate": [Function],
    "createWebhook": [Function],
    "declineInvitation": [Function],
    "delete": [Function],
    "deleteAccessRestrictions": [Function],
    "deleteAdminBranchProtection": [Function],
    "deleteBranchProtection": [Function],
    "deleteCommitComment": [Function],
    "deleteCommitSignatureProtection": [Function],
    "deleteDeployKey": [Function],
    "deleteDeployment": [Function],
    "deleteFile": [Function],
    "deleteInvitation": [Function],
    "deletePagesSite": [Function],
    "deletePullRequestReviewProtection": [Function],
    "deleteRelease": [Function],
    "deleteReleaseAsset": [Function],
    "deleteWebhook": [Function],
    "disableAutomatedSecurityFixes": [Function],
    "disableVulnerabilityAlerts": [Function],
    "downloadArchive": [Function],
    "downloadTarballArchive": [Function],
    "downloadZipballArchive": [Function],
    "enableAutomatedSecurityFixes": [Function],
    "enableVulnerabilityAlerts": [Function],
    "get": [Function],
    "getAccessRestrictions": [Function],
    "getAdminBranchProtection": [Function],
    "getAllStatusCheckContexts": [Function],
    "getAllTopics": [Function],
    "getAppsWithAccessToProtectedBranch": [Function],
    "getBranch": [Function],
    "getBranchProtection": [Function],
    "getClones": [Function],
    "getCodeFrequencyStats": [Function],
    "getCollaboratorPermissionLevel": [Function],
    "getCombinedStatusForRef": [Function],
    "getCommit": [Function],
    "getCommitActivityStats": [Function],
    "getCommitComment": [Function],
    "getCommitSignatureProtection": [Function],
    "getCommunityProfileMetrics": [Function],
    "getContent": [Function],
    "getContributorsStats": [Function],
    "getDeployKey": [Function],
    "getDeployment": [Function],
    "getDeploymentStatus": [Function],
    "getLatestPagesBuild": [Function],
    "getLatestRelease": [Function],
    "getPages": [Function],
    "getPagesBuild": [Function],
    "getParticipationStats": [Function],
    "getPullRequestReviewProtection": [Function],
    "getPunchCardStats": [Function],
    "getReadme": [Function],
    "getRelease": [Function],
    "getReleaseAsset": [Function],
    "getReleaseByTag": [Function],
    "getStatusChecksProtection": [Function],
    "getTeamsWithAccessToProtectedBranch": [Function],
    "getTopPaths": [Function],
    "getTopReferrers": [Function],
    "getUsersWithAccessToProtectedBranch": [Function],
    "getViews": [Function],
    "getWebhook": [Function],
    "getWebhookConfigForRepo": [Function],
    "listBranches": [Function],
    "listBranchesForHeadCommit": [Function],
    "listCollaborators": [Function],
    "listCommentsForCommit": [Function],
    "listCommitCommentsForRepo": [Function],
    "listCommitStatusesForRef": [Function],
    "listCommits": [Function],
    "listContributors": [Function],
    "listDeployKeys": [Function],
    "listDeploymentStatuses": [Function],
    "listDeployments": [Function],
    "listForAuthenticatedUser": [Function],
    "listForOrg": [Function],
    "listForUser": [Function],
    "listForks": [Function],
    "listInvitations": [Function],
    "listInvitationsForAuthenticatedUser": [Function],
    "listLanguages": [Function],
    "listPagesBuilds": [Function],
    "listPublic": [Function],
    "listPullRequestsAssociatedWithCommit": [Function],
    "listReleaseAssets": [Function],
    "listReleases": [Function],
    "listTags": [Function],
    "listTeams": [Function],
    "listWebhooks": [Function],
    "merge": [Function],
    "pingWebhook": [Function],
    "removeAppAccessRestrictions": [Function],
    "removeCollaborator": [Function],
    "removeStatusCheckContexts": [Function],
    "removeStatusCheckProtection": [Function],
    "removeTeamAccessRestrictions": [Function],
    "removeUserAccessRestrictions": [Function],
    "renameBranch": [Function],
    "replaceAllTopics": [Function],
    "requestPagesBuild": [Function],
    "setAdminBranchProtection": [Function],
    "setAppAccessRestrictions": [Function],
    "setStatusCheckContexts": [Function],
    "setTeamAccessRestrictions": [Function],
    "setUserAccessRestrictions": [Function],
    "testPushWebhook": [Function],
    "transfer": [Function],
    "update": [Function],
    "updateBranchProtection": [Function],
    "updateCommitComment": [Function],
    "updateInformationAboutPagesSite": [Function],
    "updateInvitation": [Function],
    "updatePullRequestReviewProtection": [Function],
    "updateRelease": [Function],
    "updateReleaseAsset": [Function],
    "updateStatusCheckPotection": [Function],
    "updateStatusCheckProtection": [Function],
    "updateWebhook": [Function],
    "updateWebhookConfigForRepo": [Function],
    "uploadReleaseAsset": [Function],
  },
  "request": [Function],
  "retry": Object {
    "retryRequest": [Function],
  },
  "search": Object {
    "code": [Function],
    "commits": [Function],
    "issuesAndPullRequests": [Function],
    "labels": [Function],
    "repos": [Function],
    "topics": [Function],
    "users": [Function],
  },
  "secretScanning": Object {
    "getAlert": [Function],
    "listAlertsForRepo": [Function],
    "updateAlert": [Function],
  },
  "teams": Object {
    "addOrUpdateMembershipForUserInOrg": [Function],
    "addOrUpdateProjectPermissionsInOrg": [Function],
    "addOrUpdateRepoPermissionsInOrg": [Function],
    "checkPermissionsForProjectInOrg": [Function],
    "checkPermissionsForRepoInOrg": [Function],
    "create": [Function],
    "createDiscussionCommentInOrg": [Function],
    "createDiscussionInOrg": [Function],
    "deleteDiscussionCommentInOrg": [Function],
    "deleteDiscussionInOrg": [Function],
    "deleteInOrg": [Function],
    "getByName": [Function],
    "getDiscussionCommentInOrg": [Function],
    "getDiscussionInOrg": [Function],
    "getMembershipForUserInOrg": [Function],
    "list": [Function],
    "listChildInOrg": [Function],
    "listDiscussionCommentsInOrg": [Function],
    "listDiscussionsInOrg": [Function],
    "listForAuthenticatedUser": [Function],
    "listMembersInOrg": [Function],
    "listPendingInvitationsInOrg": [Function],
    "listProjectsInOrg": [Function],
    "listReposInOrg": [Function],
    "removeMembershipForUserInOrg": [Function],
    "removeProjectInOrg": [Function],
    "removeRepoInOrg": [Function],
    "updateDiscussionCommentInOrg": [Function],
    "updateDiscussionInOrg": [Function],
    "updateInOrg": [Function],
  },
  "users": Object {
    "addEmailForAuthenticated": [Function],
    "block": [Function],
    "checkBlocked": [Function],
    "checkFollowingForUser": [Function],
    "checkPersonIsFollowedByAuthenticated": [Function],
    "createGpgKeyForAuthenticated": [Function],
    "createPublicSshKeyForAuthenticated": [Function],
    "deleteEmailForAuthenticated": [Function],
    "deleteGpgKeyForAuthenticated": [Function],
    "deletePublicSshKeyForAuthenticated": [Function],
    "follow": [Function],
    "getAuthenticated": [Function],
    "getByUsername": [Function],
    "getContextForUser": [Function],
    "getGpgKeyForAuthenticated": [Function],
    "getPublicSshKeyForAuthenticated": [Function],
    "list": [Function],
    "listBlockedByAuthenticated": [Function],
    "listEmailsForAuthenticated": [Function],
    "listFollowedByAuthenticated": [Function],
    "listFollowersForAuthenticatedUser": [Function],
    "listFollowersForUser": [Function],
    "listFollowingForUser": [Function],
    "listGpgKeysForAuthenticated": [Function],
    "listGpgKeysForUser": [Function],
    "listPublicEmailsForAuthenticated": [Function],
    "listPublicKeysForUser": [Function],
    "listPublicSshKeysForAuthenticated": [Function],
    "setPrimaryEmailVisibilityForAuthenticated": [Function],
    "unblock": [Function],
    "unfollow": [Function],
    "updateAuthenticated": [Function],
  },
}
`;
